src/smt.rs

```diff
@@ -1237,7 +1237,8 @@ impl SparseMerkleTree {
             ancestor_hashes.push_back(d.current_node.hash.value_as_vec());
             let pair = Arc::new(KVPair::new(
                 &d.current_node.key,
-                &d.current_node.hash.key()[PREFIX_LEAF_HASH.len() + HASH_SIZE..],
+                &d.current_node.hash.key()
+                    [PREFIX_LEAF_HASH.len() + usize::from(self.key_length)..],
```

Comment:
KeyLength has From trait for `usize` so, it would be better to use `self.key_length.into()`